### ISSUE TITLE
integer_arithmetic gives unhelpful warnings for const expressions

### ISSUE THREAD

**Issue by ssokolow** [rust-lang/rust-clippy#1761]
I use the `integer_arithmetic` lint as a reminder to explicitly decide on what kind of behaviour I want, but it also triggers in situations like this line, where the math makes the line more readable and the `*_mul` methods aren't available.

```rust
pub const MAX_SCRIPT_SIZE: u64 = 1024 * 1024; // 1 MiB
```

It should probably be limited to non-constant expressions (or split into two lints) until an alternative is stabilized.

**Comment by phansch** [comment]
This was fixed as part of https://github.com/rust-lang/rust-clippy/pull/3331 so I'm going to go ahead and close this issue.
